src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -559,18 +557,20 @@ private async Task ExecuteClusterTransactionOneByOne(List<ClusterTransactionComm
                             mergedCommands.LastChangeVector, mergedCommands.LastTombstoneEtag, mergedCommands.ModifiedCollections);
                     }
 
-                    var result = new BatchHandler.ClusterTransactionCompletionResult
+                    var result = new ClusterTransactionCompletionResult
                     {
                         Array = mergedCommands.Replies[index],
                         IndexTask = indexTask,
                     };
-                    ClusterTransactionWaiter.SetResult(options.TaskId, index, result);
+                    RachisLogIndexNotifications.NotifyListenersAbout(index, null);
+                    ServerStore.Cluster.ClusterTransactionWaiter.TrySetResult(options.TaskId, result);
```

Comment:
what if the current node doesn't contains any of the shards ?

Reply:
In a sharded database, the results are already set by the cluster state machine so `TrySetResult` will "fail" if there are shards in the node that handles the request